cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -569,15 +608,145 @@ private boolean isToolLocallyAvailable(String toolId, String version) {
+        if (tool.isEmpty()) {
+            CommandUtil.printError(errStream, "tool '" + toolId + "' is not installed.", null, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+
+        Path balaCacheDirPath = ProjectUtils.createAndGetHomeReposPath()
+                .resolve(REPOSITORIES_DIR).resolve(CENTRAL_REPOSITORY_CACHE_NAME)
+                .resolve(ProjectConstants.BALA_DIR_NAME);
+
+        for (String supportedPlatform : SUPPORTED_PLATFORMS) {
+            try {
+                version = getLatestVersionForUpdateCommand(tool.get());
+                if (tool.get().version().equals(version)) {
+                    outStream.println("tool '" + toolId + "' is already up-to-date.");
+                    CommandUtil.exitError(this.exitWhenFinish);
+                    return;
+                }
+                if (isToolLocallyAvailable(toolId, version)) {
+                    outStream.println("tool " + toolId + ":" + version + " is already available locally.");
+                } else {
+                    pullToolFromCentral(supportedPlatform, balaCacheDirPath);
+                }
+                insertToBalToolsTomlFile();
+            } catch (PackageAlreadyExistsException e) {
+                errStream.println(e.getMessage());
+                CommandUtil.exitError(this.exitWhenFinish);
+            } catch (CentralClientException | ProjectException e) {
+                CommandUtil.printError(errStream, "unexpected error occurred while pulling tool:" + e.getMessage(),
+                        null, false);
+                CommandUtil.exitError(this.exitWhenFinish);
+            }
+        }
+    }
+
+    private String getLatestVersionForUpdateCommand(BalToolsManifest.Tool tool) throws CentralClientException {
+        Settings settings;
+        try {
+            settings = RepoUtils.readSettings();
+            // Ignore Settings.toml diagnostics in the pull command
+        } catch (SettingsTomlException e) {
+            // Ignore 'Settings.toml' parsing errors and return empty Settings object
+            settings = Settings.from();
+        }
+        System.setProperty(CentralClientConstants.ENABLE_OUTPUT_STREAM, "true");
+        CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
+                initializeProxy(settings.getProxy()), settings.getProxy().username(),
+                settings.getProxy().password(), getAccessTokenOfCLI(settings));
+        List<String> versions = client.getPackageVersions(tool.org(), tool.name(), ANY_PLATFORM,
+                RepoUtils.getBallerinaVersion());
+        return getLatestVersionInSameMinor(versions, tool.version());
+    }
+
+    private String getLatestVersionInSameMinor(List<String> versions, String currentVersionStr) {
+        SemanticVersion currentVersion = SemanticVersion.from(currentVersionStr);
+        Optional<String> latestVersionInSameMinor = versions.stream().map(SemanticVersion::from)
```

Comment:
Shall we use the `compareTo` method implemented in `SemanticVersion` and simplify this?

Reply:
Cannot use due to `compareTo` of `SemanticVersion` due to it returning an enum.